Editor/CoreAPI/StateManager.cs

```diff
@@ -0,0 +1,33 @@
+﻿using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
+
+namespace Editor.CoreAPI
+{
+    public class StateManager
+    {
+        public CoreApi CoreApi { get; set; }
+        
+        public GameLiftFleetManager FleetManager { get; set; }
+        public GameLiftComputeManager ComputeManager  { get; set; }
+        
+        public IAmazonGameLiftClientWrapper GameLiftWrapper { get; private set; }
+        
+        public IAmazonGameLiftClientFactory AmazonGameLiftClientFactory { get; set; }
+        
+        public string SelectedProfile { get; set; }
+        
+        public int SelectedFleetIndex { get; set; }
+        
+        public void SetupClientFactory()
```

Comment:
This should be the constructor that accepts a CoreApi, then remove the setter for CoreApi as I don't see a reason why the state would change what file we're assigning to